### ISSUE TITLE
plotting object with season dimension = 12

### ISSUE THREAD

**Issue by marchtaylor** [flr/ggplotFL#13]
Hi there,

I think there may need to be an adjustment to the way the season dimension is handled in ggplotFL. In my example, I have 12 months in the season dimension. I believe season gets originally treated as a factor, and then converted to numeric. This seems to mess up the month order in my case (e.g. "1", "10", "11", "12", "2", "3", "4", "5", "6", "7", "8", "9"). 
Below is a fix that re-orders the levels based on their sequence in the dimnames of the FLStock object.

**direct plot with ggplotFL:**
```
    plot(stk@stock.n)
```
![ggplotfl](https://user-images.githubusercontent.com/8147731/45475397-932e3580-b73c-11e8-9b66-868815d7c6c8.png)

**adjusted version with ggplot2**:
```
    DIMNAMES <- dimnames(stk@stock.n)
    df <- as.data.frame(stk@stock.n)
    df$season <- factor(df$season, levels = DIMNAMES$season)
    df$date <- as.Date(paste(df$year, 
      as.numeric(df$season), "01", sep = "-"), 
      format = "%Y-%m-%d")
    p <- ggplot2::ggplot(data = df, 
      aes(x = date, y = data, group = age)) +
      facet_wrap(~ age, ncol = 1, scales = "free_y", 
        drop=TRUE, strip.position = "right") +
      scale_y_continuous(limits = c(0,NA)) +
      geom_line()
    print(p)
```
![ggplot2](https://user-images.githubusercontent.com/8147731/45475422-a8a35f80-b73c-11e8-89a3-3bb4d2b4b6c6.png)

**Comment by iagomosqueira** [comment]
The conversion takes place in FLCore::as.data.frame(FLQuant), with date=TRUE. The date column is of class ISOdate, and season is passed as numeric. I have added a small check for seasons being months to use the first day as season start. See here

https://github.com/flr/FLCore/blob/master/R/FLQuant.R#L1037

Does this help?

**Comment by marchtaylor** [comment]
Works perfect. I assume that since I didn't need to add any additional arguments to `plot(stk@stock.n)`, that the default is `date=TRUE`? Many thanks. Feel free to close the issue. -Marc

**Comment by iagomosqueira** [comment]
The call to as.data.frame inside ggplotFL::plot sets date=TRUE, yes, although only uses it in the x axis if length(season) > 1.

**Comment by marchtaylor** [comment]
Hi there - it looks to me as if this functionality has been lost. The season dimension is no longer correctly handled.

**Comment by iagomosqueira** [comment]
Have you reinsatalled from github? I have changed a bit the engine, now using our own geom_ function. Will check this is working as it should with the new code

**Comment by marchtaylor** [comment]
Yes, I don't know exactly when it happened, but installing one of the other packages (FLBPR?) automatically reinstalled several other dependencies as well.

**Comment by marchtaylor** [comment]
It would seem to me that `season` needs to be changed to a character, then numeric. As is, it looks like a direct conversion to numeric. Thus, this error is possibly not seen in an object with only 4 season levels, whereas with 12 the dimension order gets jumbled.

**Comment by iagomosqueira** [comment]
OK, will fix now, and will release new ggplotFL.

**Comment by iagomosqueira** [comment]
Could you possibly install the package in github and see if it has been fixed?

**Comment by marchtaylor** [comment]
Looks good now - thanks so much. One caveat: the plotting function removes empty ages / years. I think I am fine with this, but just wanted to make sure that you want this as the default.

**Comment by iagomosqueira** [comment]
Are you plotting a single FLQuant? It probably shouldn't, specially since the function offers you to plot dots at data=0 for NAs. Let me check

**Comment by marchtaylor** [comment]
Yes, a single FLQuant. The behavior you describe used to be the case, but no longer is. NAs, are converted to zeros, and dims containing all NAs are dropped completely. I can send you an example.

**Comment by marchtaylor** [comment]
here is an example of the behavior:
```
data("ple4")
stkn <- ple4@stock.n
stkn[1,] <- NaN
stkn[,ac(1981:2017)] <- NaN
plot(stkn)
```
![rplot](https://user-images.githubusercontent.com/8147731/53730985-b29b5500-3e79-11e9-97f8-171e5e3f811d.png)

**Comment by iagomosqueira** [comment]
The old code worked as expected if the break included the start or end of the time series, so the line was shown shorter. Breaking the line is a different thing. Can be done, but requires a bit more work. Working now on getting empty panels back, and having the NA dots plotted.

**Comment by marchtaylor** [comment]
Thank you for this. Sorry, I didn't know it would be this complicated. Yes,
line breaks (i.e. NaN during an otherwise non-NA time series) also appear
to be ignored (or removed before plotting).

**Comment by iagomosqueira** [comment]
Not at all, thanks for noticing. This works for me now

```
data(ple4)
x<-catch.n(ple4)
x[, ac(1965:1973)] <-NA
x[2,]<-NA
plot(x)
```
Can you confirm if it does for you?

It is still not working this way in the case when iters are involved, the lines get joined. So this issue can remain open for the time being

**Comment by iagomosqueira** [comment]
One way to achieve it might be to make geom_flquantiles to create separate geom_ribbon calls, r rathr grid_polygon ones, for each segment interrupted by NAs in the year+season dimension.

**Comment by marchtaylor** [comment]
Yes, thank you. The plot is working as before now. Until now, I have not been plotting objects with multiple iters. In that case, I would probably manually replace NAs with zeros if needed. Maybe there can be a separate argument in the plotting function for NA replacement? Cheers

**Comment by iagomosqueira** [comment]
I think changes to data are best made outside of the plot call. I will try to find a way to make the geom cut those time steps all missing.
